### ISSUE TITLE
Ansible v2 does not parse lookup('template') correctly

### ISSUE THREAD

**Issue by drybjed** [ansible/ansible#12681]
##### Issue Type:

Bug Report
##### Ansible Version:

Ansible devel
##### Ansible Configuration:

Default configuration
##### Environment:

Debian, Ubuntu
##### Summary:

Output generated by `lookup("template")` is not parsed correctly by Ansible (UnsafeProxy involved).
##### Steps To Reproduce:

A test case can be found in [separate repository](https://github.com/drybjed/ansible-lookup-test).
##### Expected Results:

```
PLAY [Test template lookup] ***************************************************

GATHERING FACTS ***************************************************************
ok: [localhost]

TASK: [lookup-test | Register the output] *************************************
ok: [localhost]

TASK: [lookup-test | Show the output] *****************************************
ok: [localhost] => {
    "var": {
        "debug_variable": "  - 'The one is surrounded'\n  - 'The two is surrounded'\n  - 'The three is surrounded'\n"
    }
}

TASK: [lookup-test | Display the list] ****************************************
ok: [localhost] => (item=The one is surrounded) => {
    "item": "The one is surrounded",
    "msg": "The one is surrounded"
}
ok: [localhost] => (item=The two is surrounded) => {
    "item": "The two is surrounded",
    "msg": "The two is surrounded"
}
ok: [localhost] => (item=The three is surrounded) => {
    "item": "The three is surrounded",
    "msg": "The three is surrounded"
}

TASK: [lookup-test | Display the static list] *********************************
ok: [localhost] => (item=aaa) => {
    "item": "aaa",
    "msg": "aaa"
}
ok: [localhost] => (item=bbb) => {
    "item": "bbb",
    "msg": "bbb"
}
```
##### Actual Results:

```
PLAY [Test template lookup] ****************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [lookup-test : Register the output] ***************************************
ok: [localhost]

TASK [lookup-test : Show the output] *******************************************
ok: [localhost] => {
    "changed": false,
    "debug_variable": "  - 'The one is surrounded'\n  - 'The two is surrounded'\n  - 'The three is surrounded'\n"
}

TASK [lookup-test : Display the list] ******************************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! an unexpected type error occurred. Error was expected string or buffer"}
...ignoring

TASK [lookup-test : Display the static list] ***********************************
fatal: [localhost]: FAILED! => {"failed": true, "msg": "ERROR! an unexpected type error occurred. Error was must be convertible to a buffer, not UnsafeProxy(str)"}
...ignoring
```

**Comment by jimi-c** [comment]
This is not related to parsing, but instead is a side-effect of wrapping lookup() results with UnsafeProxy. Currently investigating work-arounds or fixes.

**Comment by jimi-c** [comment]
Should be fixed by cd2cb178aeb799efe4aba36f93d48c95f6da3b61, just waiting for a little burn-in due to the major change involved in that commit.

**Comment by jimi-c** [comment]
# Closing This Ticket

Hi!

We believe the above commit should resolve this problem for you. This will also be included in the next major release.

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
